Vizzuality/gfw
# Feature/analysis v2

![screen shot 2018-09-21 at 19 17 07](https://user-images.githubusercontent.com/20288774/45895981-00903500-bdd3-11e8-9766-fe8a004437a4.png)

## Overview

Reimplements the GFW analysis feature in the new map. This brings with it a series of improvements.

- The navigation has been updated to move smoother through the create/cancel analysis flow
- The file upload tool has been improved with real error feedback on file types, sizes, and ogr2ogr errors.
- One click analysis is available on boundary and interactive layers from the route analysis menu
- Popup now have working analyze buttons throughout
- Dynamic fetch and response building within the analysis services (rather 😎 )
- Drawing is now simpler and smoother
- The whole fetching and flow is faster due to reduced code base.

## Demo

You can review on staging, or but pulling down this branch. The feature is too complex to list all options here.

## Notes

Many small bug fixes are included in this branch also that we discovered along the way. Mostly styles but other are:
- embeds now work for the map
- SEO and metadata passing from rails to app (with dynamic SEO calls in rails and synced with React)
- HMR is improved with a bug fix for infinite refresh with React 16 using universal component

## Testing

- Make a one click analysis on each boundary layer
- Make a draw analysis anywhere on the map
- Upload a shape file
- Upload an incorrect file to see errors
- Toggle layers in analysis mode to see updates
- Change time line and threshold of each layer to see syncing.
